src/main/java/org/broadinstitute/hellbender/utils/clipping/ClippingOp.java

```diff
@@ -365,6 +366,14 @@ private GATKRead applyHARDCLIP_BASES(final GATKRead read, final int start, final
         // the cigar may force a shift left or right (or both) in case we are left with insertions
         // starting or ending the read after applying the hard clip on start/stop.
         final int newLength = read.getLength() - (stop - start + 1) - cigarShift.shiftFromStart - cigarShift.shiftFromEnd;
+
+        // If the new read is going to be empty, return an empty read now. This avoids initializing the new
+        // read with invalid values below in certain cases (such as a negative alignment start).
+        // See https://github.com/broadinstitute/gatk/issues/3466
+        if ( newLength == 0 ) {
+            return ReadUtils.emptyRead(read);
```

Comment:
Right, it seems like this bug really was only for empty reads which caused the calculation of the new offset to break.